File: 23.tell-dont-ask/src/java/src/test/java/com/telldontask/doubles/TestOrderRepository.java
package com.telldontask.doubles;

import com.telldontask.domain.Order;
import com.telldontask.repository.OrderRepository;

import java.util.ArrayList;
import java.util.List;

public class TestOrderRepository implements OrderRepository {
    private Order insertedOrder;
    private final List<Order> orders = new ArrayList<>();

    @Override
    public void save(Order order) {
        this.insertedOrder = order;
    }

    @Override
    public Order getById(int orderId) {
        return orders.stream()
                .filter(o -> o.getId() == orderId)
                .findFirst()
                .orElse(null);
    }

    public Order getSavedOrder() {
        return insertedOrder;
    }

    public void addOrder(Order order) {
        orders.add(order);
    }
}

File: 23.tell-dont-ask/src/typescript/tests/doubles/testOrderRepository.ts
import { Order } from '../../domain/order';
import { OrderRepository } from '../../repository/orderRepository';

export class TestOrderRepository implements OrderRepository {
  private insertedOrder: Order | undefined;
  private readonly orders: Order[] = [];

  save(order: Order): void {
    this.insertedOrder = order;
  }

  getById(orderId: number): Order | undefined {
    return this.orders.find((o) => o.id === orderId);
  }

  getSavedOrder(): Order | undefined {
    return this.insertedOrder;
  }

  addOrder(order: Order): void {
    this.orders.push(order);
  }
}
